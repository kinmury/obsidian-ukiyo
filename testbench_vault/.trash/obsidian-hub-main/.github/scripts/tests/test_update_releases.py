from approvaltests import verify_as_json

from update_releases import collate_authors
from tests.helpers_for_testing import \
    get_processed_saved_sample_data_for_plugin, get_processed_saved_sample_data_for_theme


def test_process_authors() -> None:
    theme = get_processed_saved_sample_data_for_theme("Christmas")
    themes = [theme]

    plugin = get_processed_saved_sample_data_for_plugin("nldates-obsidian")
    plugins = [plugin]

    all_authors = collate_authors(themes, plugins)
    verify_as_json(all_authors)
