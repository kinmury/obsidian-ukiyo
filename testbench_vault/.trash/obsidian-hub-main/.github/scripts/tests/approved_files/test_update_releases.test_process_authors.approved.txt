{
    "argenos": {
        "author": "Argentina Ortega Sainz",
        "plugins": [
            "[[nldates-obsidian|Natural Language Dates]]"
        ],
        "user": "argenos",
        "website": "https://argentinaos.com/"
    },
    "deathau": {
        "author": "deathau",
        "themes": [
            "[[Christmas]]"
        ],
        "user": "deathau"
    }
}
